pkg/bot/lark.go

```diff
@@ -0,0 +1,225 @@
+func (l *LarkBot) Execute(e map[string]interface{}) {
+	event, err := AccessAndTypeCastToMap(Event, e)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToMap: %s", err.Error())
+		return
+	}
+
+	chatType, err := AccessAndTypeCastToString(ChatType, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+
+	text, err := AccessAndTypeCastToString(Text, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+
+	executor := execute.NewDefaultExecutor(text, l.AllowKubectl, l.RestrictAccess, l.DefaultNamespace,
+		l.ClusterName, config.LarkBot, "", true)
+	response := executor.Execute()
+
+	if chatType == "group" {
+		chatID, err := AccessAndTypeCastToString(OpenChatID, event)
+		if err != nil {
+			log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+			return
+		}
+
+		l.LarkClient.SendTextMessage(ChatID, chatID, response)
+		return
+	}
+	openID, err := AccessAndTypeCastToString(OpenID, event)
+	if err != nil {
+		log.Errorf("AccessAndTypeCastToString: %s", err.Error())
+		return
+	}
+	l.LarkClient.SendTextMessage(OpenID, openID, response)
+}
+
+// Start starts the lark server and listens for lark messages
+func (l *LarkBot) Start() {
+	// See: https://open.larksuite.com/document/ukTMukTMukTM/ukjNxYjL5YTM24SO2EjN
+	// message
+	larkConf := l.LarkClient.Conf
+	event.SetTypeCallback(larkConf, Message, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.Execute(e)
+		return nil
+	})
+
+	// add_bot
+	event.SetTypeCallback(larkConf, AddBot, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.SayHello(e)
+		return nil
+	})
+
+	// p2p_chat_create
+	event.SetTypeCallback(larkConf, P2pChatCreate, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.SayHello(e)
+		return nil
+	})
+
+	// add_user_to_chat
+	event.SetTypeCallback(larkConf, AddUserToChat, func(ctx *core.Context, e map[string]interface{}) error {
+		log.Infof(ctx.GetRequestID())
+		log.Infof(tools.Prettify(e))
+		go l.SayHello(e)
+		return nil
+	})
+
+	eventhttpserver.Register(l.MessagePath, larkConf)
+	log.Infof("Started lark server on port %d", l.Port)
+	log.Errorf("Error in lark server. %v", http.ListenAndServe(fmt.Sprintf(":%d", l.Port), nil))
+}
+
+// SayHello send welcome message to new added users
+func (l *LarkBot) SayHello(e map[string]interface{}) error {
+	event, err := AccessAndTypeCastToMap(Event, e)
+	if err != nil {
+		return err
+	}
+
+	larkUsers, ok := event[Users]
+	if !ok {
+		return fmt.Errorf("%s not found", larkUsers)
+	}
+	users, ok := larkUsers.([]interface{})
+	if !ok {
+		user, ok := event[Users].(interface{})
+		if !ok {
+			return fmt.Errorf("typecast error - failed to convert into interface{}")
+		}
+		users = append(users, user)
+	}
```

Comment:
```suggestion
	if !ok {
	        return fmt.Errorf("Invalid users format received in response. Failed to convert users into []interface{}")
	}
	users = append(users, user)
```
@Lion916 Are we doing repeated operations here? L187 and L189 does the same thing